plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GestureState.kt

```diff
@@ -0,0 +1,38 @@
+package com.mapbox.maps.plugin.gestures
+
+import com.mapbox.android.gestures.AndroidGesturesManager
+
+internal class GestureState(private val gesturesManager: AndroidGesturesManager) {
+  enum class Type {
+    DoubleTap,
+    Scale,
+    ScaleQuickZoom,
+    Shove,
+  }
+
+  private val savedGestureEnabledMap = mutableMapOf<Type, Boolean>()
```

Comment:
nit:
btw, if you have fixed size of types, maybe replace map with array and get value by index? it will be the same speed, but you can save a little space